server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/AddDefaultAttributesMessageHook.java

```diff
@@ -53,11 +56,11 @@ public HookResult onMessage(SMTPSession session, Mail mail) {
             mailImpl.setRemoteHost(session.getRemoteAddress().getHostName());
             mailImpl.setRemoteAddr(session.getRemoteAddress().getAddress().getHostAddress());
             if (session.getUser() != null) {
-                mail.setAttribute(Mail.SMTP_AUTH_USER_ATTRIBUTE_NAME, session.getUser());
+                mail.setAttribute(new Attribute(Mail.SMTP_AUTH_USER_ATTRIBUTE_NAME, AttributeValue.of(session.getUser())));
```

Comment:
do we hate developers to require them to wrap values by themselves ?
It doesn't add any safety, it's just more painful to type (and to read).
Could we think about having an easier API ?

Reply:
Attribute::convertToAttribute can at least help with AttributeValue boxing

Here:

```
mail.setAttribute(Attribute.convertToAttribute(Mail.SMTP_AUTH_USER_ATTRIBUTE_NAME, session.getUser()));
```

Reply:
We require from the developers to have a clear mind about the values he/she is dealing with by enforcing a specific type as soon as possible.

The sooner a value is strongly typed, the less error prone code you are able to write.

It's a gift given to the developer, he/she can do (as in this example) only type it at the end, but he/she won't use the given possibilities.